src/walletNode.ts

```diff
@@ -275,18 +276,43 @@ export class WalletNode {
       response.content.genesisBlockIdentifier.hash,
       'hex',
     )
-    const walletGenesisHeader = BlockHeaderSerde.deserialize(
-      networkDefinition.genesis.header,
-      this.strategy,
-    )
 
-    if (walletGenesisHeader.hash.equals(nodeGenesisHash)) {
+    let fishHashContext: FishHashContext | undefined = undefined
+    if (!this.network.consensus.isNeverActive('enableFishHash')) {
+      fishHashContext = new FishHashContext(false)
+    }
+
+    const blockHasher = new BlockHasher({
+      consensus: this.network.consensus,
+      context: fishHashContext,
+    })
+
+    const genesisHeader = networkDefinition.genesis.header
+
+    // TODO: This copies code from BlockHeaderSerde (https://github.com/iron-fish/ironfish/blob/master/ironfish/src/primitives/blockheader.ts)
+    // Refactor to use the same code once it is exported by the SDK
+    const walletGenesisHash = blockHasher.hashHeader({
+      sequence: Number(genesisHeader.sequence),
+      previousBlockHash: Buffer.from(
+        BlockHashSerdeInstance.deserialize(genesisHeader.previousBlockHash),
+      ),
+      noteCommitment: genesisHeader.noteCommitment,
+      transactionCommitment: genesisHeader.transactionCommitment,
+      target: new Target(genesisHeader.target),
+      randomness: BigInt(genesisHeader.randomness),
+      timestamp: new Date(genesisHeader.timestamp),
+      graffiti: Buffer.from(
+        GraffitiSerdeInstance.deserialize(genesisHeader.graffiti),
+      ),
+    })
```

Comment:
this would be nicer if the wallet CLI  at least had a method to just convert a header definition to a header and then used that here to make it clear that's what's happening. The function would be `headerDefinitionToHeader(definition: NetworkDefinition['genesis']['header']) => BlockHeader`